xds/src/main/java/io/grpc/xds/ClusterImplLoadBalancer.java

```diff
@@ -213,36 +214,42 @@ public void updateBalancingState(ConnectivityState newState, SubchannelPicker ne
     @Override
     public Subchannel createSubchannel(CreateSubchannelArgs args) {
       List<EquivalentAddressGroup> addresses = withAdditionalAttributes(args.getAddresses());
-      Locality locality = args.getAddresses().get(0).getAttributes().get(
-          InternalXdsAttributes.ATTR_LOCALITY);  // all addresses should be in the same locality
-      String localityName = args.getAddresses().get(0).getAttributes().get(
-          InternalXdsAttributes.ATTR_LOCALITY_NAME);
-      // Endpoint addresses resolved by ClusterResolverLoadBalancer should always contain
-      // attributes with its locality, including endpoints in LOGICAL_DNS clusters.
-      // In case of not (which really shouldn't), loads are aggregated under an empty locality.
-      if (locality == null) {
-        locality = Locality.create("", "", "");
-        localityName = "";
-      }
-      final ClusterLocalityStats localityStats =
-          (lrsServerInfo == null)
-              ? null
-              : xdsClient.addClusterLocalityStats(lrsServerInfo, cluster,
-              edsServiceName, locality);
-
+      // This clusterLocality ideally should not be utilized. We derive locality
+      // information from the first address, even prior to the subchannel becoming ready.
+      // This is primarily for the purpose of facilitating load reporting in the pre-READY
```

Comment:
There's no load reporting pre-READY. But an LB API could return the subchannel even before it is ready, and PF does this. We generally won't end up reporting load for such picks because the channel will ignore the selected (not-ready) subchannel, but we needed to handle the case.

Reply:
Did you miss pushing the update?

Reply:
Discussed offline, we both are able to see the updated comment.